traveler-ios-ui/TravelerKitUI/ViewControllers/CancellationViewController.swift

```diff
@@ -15,50 +15,127 @@ protocol CancellationViewControllerDelegate: class {
 
     override func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return 2
     }
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return quote?.products.count ?? 0
+        if section == 0 {
+            return (quote?.products.count ?? 0) + 1
+        } else {
+            return (quote?.cancellationReasons.count ?? 0)
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if section == 1 {
+            return "Please select a reason for cancellation"
+        } else {
+            return nil
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        if section == 0 {
+            return " "
+        } else {
+            return nil
+        }
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "productCancelCell", for: indexPath) as! InfoCell
+        if indexPath.section == 0 {
+            if indexPath.row >= (quote?.products.count ?? 0) {
+                let cell = tableView.dequeueReusableCell(withIdentifier: "productCancelSummaryCell", for: indexPath) as! InfoCell
+                cell.secondValueLabel?.text = quote?.totalRefund.localizedDescriptionInBaseCurrency
+
+                return cell
+            } else {
+                let cell = tableView.dequeueReusableCell(withIdentifier: "productCancelCell", for: indexPath) as! InfoCell
+
+                cell.titleLabel.text = quote?.products[indexPath.row].title
+                let percentageFee = quote!.percentageFee(product: quote!.products[indexPath.row])
+                cell.valueLabel.text = "Cancellation Fee \(percentageFee)%"
+                cell.secondValueLabel?.text = quote?.products[indexPath.row].cancellationCharge.localizedDescriptionInBaseCurrency
+
+                return cell
+            }
+        } else {
+            let cell = tableView.dequeueReusableCell(withIdentifier: "cancelReasonCell", for: indexPath) as! InfoCell
+            cell.titleLabel.text = quote?.cancellationReasons[indexPath.row].value
+            cell.accessoryType = indexPath == selectedReasonIndexPath ? .checkmark : .none
+
+            return cell
+        }
+    }
+
+    // MARK: UITableViewDelegate
 
-        cell.titleLabel.text = quote?.products[indexPath.row].title
-        let percentageFee = quote!.percentageFee(product: quote!.products[indexPath.row])
-        cell.valueLabel.text = "Cancellation Fee \(percentageFee)%"
-        cell.secondValueLabel?.text = quote?.products[indexPath.row].cancellationCharge.localizedDescriptionInBaseCurrency
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard indexPath.section == 1 else {
```

Comment:
1) you can remove interactivity for individual cells in the storyboard so that you don't have to guard for this. 
2) This is another case when a switch is a better option. Running a switch on the section would achieve the same outcome you have with your guard statement with the advantage that if in the future we need to add more interactive sections to this table, you only need to add a case rather than rewriting this method.